File: minimum_waiting_time/minimum_waiting_time.java
import java.util.*;

public class MinimumWaitingTime {
    public static int minimumWaitingTime(int[] queries) {
        Arrays.sort(queries);
        int totalWaitingTime = 0;

        for (int i = 0; i < queries.length; i++) {
            int queriesLeft = queries.length - (i + 1);
            totalWaitingTime += queries[i] * queriesLeft;
        }

        return totalWaitingTime;
    }

    public static void main(String[] args) {
        int[] queries = {3, 2, 1, 2, 6};
        System.out.println(minimumWaitingTime(queries));
    }
}

File: minimum_waiting_time/minimum_waiting_time.py
def minimum_waiting_time(queries):
  queries.sort()
  total_waiting_time = 0

  for idx, duration in enumerate(queries):
    queries_left = len(queries) - (idx + 1)
    total_waiting_time += queries_left * duration

  return total_waiting_time

print(minimum_waiting_time([3, 2, 1, 2, 6]))